plugins/out_s3/s3.c

```diff
@@ -1816,6 +1855,7 @@ static void cb_s3_upload(struct flb_config *config, void *data)
         if (complete == FLB_TRUE) {
             m_upload->upload_state = MULTIPART_UPLOAD_STATE_COMPLETE_IN_PROGRESS;
             mk_list_del(&m_upload->_head);
+            // todo: if multipart compressed upload chunk size < 5MB, data will upload but complete_multipart_upload will report error and kill fluent-bit. Need to work on it in next revision.
```

Comment:
If the PR is still in progress but you want feedback- GitHub has this feature called draft PRs, this is ideal for that

Reply:
got it, will do next time